borrowing/views.py

```diff
@@ -1,3 +1,84 @@
-from django.shortcuts import render
+from django.db import transaction
+from django.utils import timezone
+from rest_framework import generics, status
+from rest_framework.decorators import api_view
+from rest_framework.request import Request
+from rest_framework.response import Response
 
-# Create your views here.
+from borrowing.models import Borrowing, Book
+from borrowing.serializers import BorrowingSerializer
+
+
+class BorrowingListCreateAPIView(generics.ListCreateAPIView):
+    queryset = Borrowing.objects.all()
+    serializer_class = BorrowingSerializer
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        user_id = self.request.query_params.get("user_id")
+        is_active = self.request.query_params.get("is_active")
+
+        if user_id is not None:
+            queryset = queryset.filter(user=user_id)
+
+        if is_active is not None:
+            is_active = is_active.lower() in ("true", "1")
+            if is_active:
+                queryset = queryset.filter(actual_return_date__isnull=True)
+            else:
+                queryset = queryset.filter(actual_return_date__isnull=False)
+
+        return queryset
+
+    def create(self, request, *args, **kwargs):
+        book_id = request.data.get("book")
+        try:
+            book = Book.objects.get(id=book_id)
+        except Book.DoesNotExist:
+            return Response(
+                {"error": "Book not found"},
+                status=status.HTTP_404_NOT_FOUND
+            )
+
+        book.inventory -= 1
+        book.save()
+
+        return super().create(request, *args, **kwargs)
+
+
+class BorrowingDetailAPIView(generics.RetrieveAPIView):
+    queryset = Borrowing.objects.all()
+    serializer_class = BorrowingSerializer
+

```

Comment:
add permissions

Reply:
its going to be the next step in next PR